QueueArray.java

```diff
@@ -0,0 +1,194 @@
+
+/**
+ *
+ * @author cuong.hq1
+ */
+public class QueueArray {
+
+    private int capacity;
+    private int[] queue;
+
+    // Indicate front indicator of queue, alway point at non-value index,
+    // next index of front indicator will be the front value of queue.
+    private int front;
+
+    // Indicate rear (tail) of queue.
+    private int rear;
+
+    /**
+     * Initialize queue with the current capacity of the queue to be k. **
+     * Instead of initialize queue with capacity of k, add a extra element in
+     * queue array to prevent confuse situation when rear indicator and front
+     * one at a same position, this will bring.
+     *
+     */
+    public void init(int capacity) {
+        this.capacity = capacity + 1;
+        queue = new int[capacity];
+        front = 0;
+        rear = 0;
+    }
+
+    public QueueArray(int capacity) {
+        this.init(capacity);
+    }
+
+    /**
+     * Insert an element into the circular queue. Return true if the operation
+     * is successful.
+     *
+     * Example with capacity = 4;
+     *
+     * [INIT]
+     *
+     * [ ] [ ] [ ] [ ] [ ]
+     * | |
+     * F R
+     *
+     * [ENQUEUE X]
+     *
+     * [ ] [X] [ ] [ ] [ ] | | F R
+     */
+    public boolean enQueue(int value) {
+        if (isFull()) {
+            return false;
+        }
+
+        rear = (rear + 1) % capacity;
+        queue[rear] = value;
+        return true;
+    }
+
+    /**
+     * Delete an element from the circular queue. Return true if the operation
+     * is successful.
+     *
+     * EXAMPLE: [a] [ ] [b] [x] [y] | | R F
+     *
+     * [DEQUEUE] [a] [ ] [ ] [x] [y] | | R F
+     *
+     */
+    public boolean deQueue() {
+
+        if (isEmpty()) {
+            return false;
+        }
+
+        front = (front + 1) % capacity;
+        return true;
+    }
+
+    /**
+     * Get the front item from the queue.
+     */
+    public int front() {
+        if (isEmpty()) {
+            return -1;
+        }
+        result(queue[(front + 1) % capacity] + "");
```

Comment:
Is this `result` method used for debugging purpose? You should create test cases to test your code instead of putting redundant methods into the functions.